src/combinatorics/simple_graph/basic.lean

```diff
@@ -638,6 +648,51 @@ lemma delete_edges_eq_inter_edge_set (s : set (sym2 V)) :
   G.delete_edges s = G.delete_edges (s ∩ G.edge_set) :=
 by { ext, simp [imp_false] { contextual := tt } }
 
+lemma delete_edges_sdiff_eq_of_le {H : simple_graph V} (h : H ≤ G) :
+  G.delete_edges (G.edge_set \ H.edge_set) = H :=
+by { ext v w, split; simp [@h v w] { contextual := tt } }
+
+lemma edge_set_delete_edges (s : set (sym2 V)) :
+  (G.delete_edges s).edge_set = G.edge_set \ s :=
+by { ext e, refine sym2.ind _ e, simp }
+
+lemma edge_finset_delete_edges [fintype V] [decidable_eq V] [decidable_rel G.adj]
+  (s : finset (sym2 V)) [decidable_rel (G.delete_edges s).adj] :
+  (G.delete_edges s).edge_finset = G.edge_finset \ s :=
+by { ext e, simp [edge_set_delete_edges] }
+
+section delete_far
+variables (G) [linear_ordered_field 𝕜] [fintype V] [decidable_eq V] [decidable_rel G.adj]
+  {p : simple_graph V → Prop} {ε δ : 𝕜}
```

Comment:
This `ε` should be at least 1, right? Maybe use another letter, even if it is just a scaled-up analogue of the `ε` used later?